full/coupler_main.F90

```diff
@@ -615,79 +612,56 @@ program coupler_main
 
-      ! This could be a point where the model is serialized; This calls on all ice PEs
-      if (.not.concurrent_ice) call coupler_exchange_fast_to_slow_ice(Ice, coupler_clocks, &
-                                                                      set_ice_current_pelist=.True.)
+      Time = Time_atmos
 
-      !   ------ slow-ice model ------
+      !> end atm_clock_1
+      call fms_mpp_clock_end(coupler_clocks%atm)
 
-      ! This call occurs on whichever PEs handle the slow ice processess.
-      if (Ice%slow_ice_PE .and. .not.combined_ice_and_ocean) then
-        if (slow_ice_with_ocean) call fms_mpp_set_current_pelist(Ice%slow_pelist)
-        call fms_mpp_clock_begin(coupler_clocks%update_ice_model_slow_slow)
-        call update_ice_model_slow(Ice)
+    endif atm_pe_block
 
-        call fms_mpp_clock_begin(coupler_clocks%flux_ice_to_ocean_stocks)
-        call flux_ice_to_ocean_stocks(Ice)
-        call fms_mpp_clock_end(coupler_clocks%flux_ice_to_ocean_stocks)
-        call fms_mpp_clock_end(coupler_clocks%update_ice_model_slow_slow)
-      endif
+    !> Ice is still using ATM pelist and need to be included in ATM clock
+    !> ATM clock is used for load-balancing the coupled models
+    start_atm_clock2: if(Atm%pe) then
+      call fms_mpp_clock_begin(coupler_clocks%atm)
+    end if start_atm_clock2
 
+    if (do_ice .and. Ice%pe) then
+      if (Ice%fast_ice_PE) call coupler_unpack_land_ice_boundary(Ice, Land_ice_boundary, coupler_clocks)
+      !> This could be a point where the model is serialized; This calls on all ice PEs
+      if (.not.concurrent_ice) call coupler_exchange_fast_to_slow_ice(Ice, coupler_clocks, &
+                               set_ice_current_pelist=.True.)
+      !> slow-ice model
+      !! This call occurs on whichever PEs handle the slow ice processess.
+      if (Ice%slow_ice_PE .and. .not.combined_ice_and_ocean) &
+          call coupler_update_ice_model_slow_and_stocks(Ice, coupler_clocks)
       if (do_chksum) call coupler_chksum_obj%get_slow_ice_chksums('update_ice_slow+', nc)
-     endif  ! End of Ice%pe block
+    endif  ! End of Ice%pe block
 
-     if(Atm%pe) then
-        call fms_mpp_set_current_pelist(Atm%pelist)
-        call fms_mpp_clock_end(coupler_clocks%atm)
-     endif
+    end_atm_clock2: if(Atm%pe) then
+      call fms_mpp_set_current_pelist(Atm%pelist)
+      call fms_mpp_clock_end(coupler_clocks%atm)
+    endif end_atm_clock2
 
     ! Update Ice_ocean_boundary using the newly calculated fluxes.
     if ((concurrent_ice .or. .not.use_lag_fluxes) .and. .not.combined_ice_and_ocean) then
       !this could serialize unless slow_ice_with_ocean is true.
       if ((.not.do_ice) .or. (.not.slow_ice_with_ocean)) call fms_mpp_set_current_pelist()
-      call coupler_flux_ice_to_ocean(Ice, Ocean, Ice_ocean_boundary, coupler_clocks, &
-        slow_ice_ocean_pelist=slow_ice_ocean_pelist, set_current_slow_ice_ocean_pelist=.True.)
+      if (Ice%slow_ice_PE .or. Ocean%is_ocean_pe) &
```

Comment:
I thought I had the fix in the previous PR, but I must not have.  The if statement should be here.